red_queen/games/applications/hidden_shift.py

```diff
@@ -0,0 +1,138 @@
+# ---------------------------------------------------------------------------------------------
+# Part of Red Queen Project.  This file is distributed under the Apache 2.0 License.
+# See accompanying file /LICENSE for details.
+# ---------------------------------------------------------------------------------------------
+
+# Based on the code by Team Hidden Name, Qiskit Hackathon Korea 2022
+
+# github repository:
+# https://github.com/Team-Hidden-Name/hidden-shift-problem/blob/main/hidden-shift-problem.ipynb
```

Comment:
I don't see any license information in this notebook or in the hidden-shift-problem repo is this the same code as https://github.com/qiskit-community/qiskit-textbook/blob/main/content/ch-algorithms/hidden-shift-problem.ipynb if so copying code from this isn't a problem because they contributed it to an apache licensed repo (although we might want to point to the qiskit textbook notebook instead so there is no ambiguity). If it's different then you'll need to get permission from the authors to modify/reuse their work in an apache licensed project or get them to license their code under a compatible open source license.